proxy-node-shared/src/main/java/uk/gov/ida/notification/exceptions/proxy/MetatronResponseException.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.ida.notification.exceptions.proxy;
+
+import org.slf4j.MDC;
+import uk.gov.ida.common.ExceptionType;
+import uk.gov.ida.exceptions.ApplicationException;
+import uk.gov.ida.notification.exceptions.mappers.BaseJsonErrorResponseRuntimeException;
+import uk.gov.ida.notification.shared.logging.ProxyNodeMDCKey;
+
+import javax.ws.rs.core.Response;
+
+public class MetatronResponseException extends BaseJsonErrorResponseRuntimeException {
+
+    public static final String ERROR_MESSAGE_FORMAT = "Error making request to Metatron with entityId '%s'";
+
+    public MetatronResponseException(Throwable cause, String entityId) {
+        super(String.format(ERROR_MESSAGE_FORMAT, entityId), cause);
+        MDC.put(ProxyNodeMDCKey.EIDAS_ENTITY_ID.name(), entityId);
+    }
+
+    @Override
+    public Response.Status getResponseStatus() {
+        ApplicationException cause = (ApplicationException) this.getCause();
+        return cause.getExceptionType() == ExceptionType.CLIENT_ERROR ?
```

Comment:
```suggestion
        return ExceptionType.CLIENT_ERROR == cause.getExceptionType() ?
```

Reply:
To avoid a potential NPE, happy to be be proven that it's not possible

Reply:
Both sides of the expression are going to be evaluated either way I think. Not sure it would cause a NPE though? An `ApplicationException` can't be instantiated without an exception type being provided, and even if it was provided a null, `null == ExceptionType.CLIENT_ERROR` is still a valid comparison.

Reply:
You're right, let's keep it as is